src/eureka/S2_calibrations/s2_calibrate.py

```diff
@@ -232,15 +232,21 @@ def run_eurekaS2(self, filename, meta, log):
             Fragmented code to allow reuse of code between spectral and image
             analysis.
         '''
-        
-        if meta.slit_y_low is not None:
-            # Controls the cross-dispersion extraction
+
+        if hasattr(meta, 'slit_y_low') and meta.slit_y_low is not None:
+            #  NIRSpec subarray lower bound in cross-dispersion direction
             self.assign_wcs.slit_y_low = meta.slit_y_low
 
-        if meta.slit_y_high is not None:
-            # Controls the cross-dispersion extraction
+        if hasattr(meta, 'slit_y_high') and meta.slit_y_high is not None:
+            #  NIRSpec subarray upper bound in cross-dispersion direction
             self.assign_wcs.slit_y_high = meta.slit_y_high
 
+        if hasattr(meta, 'tsgrism_extract_height') and \
+           meta.tsgrism_extract_height is not None:
+            # NIRCam grism subarray height in cross-dispersion direction
+            self.extract_2d.tsgrism_extract_height = \
+                meta.tsgrism_extract_height
```

Comment:
Thanks for adding this feature - recent GTO data analyses have seemingly shown the extra pixels to be important